docs/dev_notes/cloud_infrastructure_design.md

```
@@ -0,0 +1,74 @@
+Start with Topology A to test cloud setup for MVP. After successful, migrate to Topology B.
+
+Rationale:
+1. Fastest path to cloud MVP.
+2. This process allows for supporting both options, giving users the flexbility to choose which one suits their needs. 
+3. Topology B is significantly more complex. Requires design for networking/communication and storage.
+
+## Known Risks
+The main risk is lots of refactoring to avoid docker-in-docker as serverless computing options do not natively support docker-in-docker, which would cause vendor lock-in.
+
+## Comparison to Other Similar Agent Harnesses
+| Platform | Architecture | |
```

markdownlint-cli2 (0.21.0)

[warning] Tables should be surrounded by blank lines

(MD058, blanks-around-tables)